### PR TITLE
fix the `./gradlew :aws:sdk:assemble` command and aws-config tests on Windows

### PR DESCRIPTION
**Pull request by Velfi**
tracked in issue #3518 

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt [hunk 1]
```diff
@@ -624,7 +624,13 @@ class RustWriter private constructor(
         init {
             expressionStart = '#'
             if (filename.endsWith(".rs")) {
-                require(namespace.startsWith("crate") || filename.startsWith("tests/") || filename == "build.rs") {
+                require(
+                    namespace.startsWith("crate") ||
+                    // Unix systems
+                    filename.startsWith("tests/") ||
```
Comment (aajtodd) [inline codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt:630]: Could you not use `File.separator` and collpase unix/windows branch?

**Comment by Velfi** [comment]
In case you're interested in the test failures that I'm currently ignoring:

```
failures:

---- default_provider::credentials::test::eks_pod_identity_no_token_file stdout ----
To see full logs from this test set VERBOSE_TEST_LOGS=true
test case: ecs-pod-identity-no-token-file. fail to load credentials from EKS Pod Identity without a token file
thread 'default_provider::credentials::test::eks_pod_identity_no_token_file' panicked at src\test_case.rs:380:14:
`an error occurred while loading credentials: The system cannot find the path specified. (os error 3) (ProviderError(ProviderError { source: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." } }))` did not contain `No such file or directory (os error 2)`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- default_provider::credentials::test::sso_no_token_file stdout ----
To see full logs from this test set VERBOSE_TEST_LOGS=true
test case: sso-role. load creds from SSO role
thread 'default_provider::credentials::test::sso_no_token_file' panicked at src\test_case.rs:380:14:
`an error occurred while loading credentials: an error occurred while loading credentials: failed to read `/home\.aws/sso/cache\dace00cba5f8355ec9d274ceb2bcebdfbeed0e12.json`: The system cannot find the file specified. (os error 2) (ProviderError(ProviderError { source: ProviderError(ProviderError { source: IoError { what: "read", path: "/home\\.aws/sso/cache\\dace00cba5f8355ec9d274ceb2bcebdfbeed0e12.json", source: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." } } }) }))` did not contain `failed to read `/home/.aws/sso/cache/dace00cba5f8355ec9d274ceb2bcebdfbeed0e12.json`: No such file or directory (os error 2)`

---- credential_process::test::test_credential_process_no_expiry stdout ----
thread 'credential_process::test::test_credential_process_no_expiry' panicked at src\credential_process.rs:260:58:
valid creds: ProviderError(ProviderError { source: "Error retrieving credentials from external process, could not parse response: invalid JSON in response: Error at offset 0: unexpected token '''. Expected one of '{', '[', '\"', 'null', 'true', 'false', <number>" })

---- credential_process::test::test_credential_process stdout ----
thread 'credential_process::test::test_credential_process' panicked at src\credential_process.rs:239:58:
valid creds: ProviderError(ProviderError { source: "Error retrieving credentials from external process, could not parse response: invalid JSON in response: Error at offset 0: unexpected token '''. Expected one of '{', '[', '\"', 'null', 'true', 'false', <number>" })

---- imds::credentials::test::external_timeout_during_credentials_refresh_should_yield_last_retrieved_credentials stdout ----
2024-03-27T19:33:56.461589Z DEBUG aws_config::imds::credentials: loading credentials from IMDS
2024-03-27T19:33:56.462066Z DEBUG aws_config::imds::client::token: IMDS token cache miss
thread 'imds::credentials::test::external_timeout_during_credentials_refresh_should_yield_last_retrieved_credentials' panicked at src\imds\credentials.rs:479:22:
provide_credentials completed before timeout future

---- imds::client::test::one_second_connect_timeout stdout ----
2024-03-27T19:33:56.458119Z DEBUG aws_config::imds::client::token: IMDS token cache miss
thread 'imds::client::test::one_second_connect_timeout' panicked at src\imds\client.rs:1006:22:
wrong error, expected construction failure with TimedOutError inside: failed to load IMDS session token: dispatch failure: io error: error trying to connect: tcp connect error: A socket operation was attempted to an unreachable network. (os error 10051): tcp connect error: A socket operation was attempted to an unreachable network. (os error 10051): A socket operation was attempted to an unreachable network. (os error 10051) (FailedToLoadToken(FailedToLoadToken { source: DispatchFailure(DispatchFailure { source: ConnectorError { kind: Io, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 10051, kind: NetworkUnreachable, message: "A socket operation was attempted to an unreachable network." })), connection: Unknown } }) }))

---- profile::credentials::sso_tests::create_inner_provider_exactly_once stdout ----
2024-03-27T19:33:56.477426Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
2024-03-27T19:33:56.477450Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/config"      
2024-03-27T19:33:56.477468Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/config" after="/home\\.aws\\config"    
2024-03-27T19:33:56.477478Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file not found path=~/.aws/config
2024-03-27T19:33:56.477483Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\config") size=0
2024-03-27T19:33:56.477502Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/credentials"
2024-03-27T19:33:56.477510Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/credentials" after="/home\\.aws\\credentials"
2024-03-27T19:33:56.477517Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file not found path=~/.aws/credentials
2024-03-27T19:33:56.477522Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\credentials") size=0
thread 'profile::credentials::sso_tests::create_inner_provider_exactly_once' panicked at src\profile\credentials.rs:707:64:
called `Result::unwrap()` on an `Err` value: CredentialsNotLoaded(CredentialsNotLoaded { source: None })

---- profile::parser::source::tests::programmatically_include_default_files stdout ----
2024-03-27T19:33:56.501417Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
2024-03-27T19:33:56.501496Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: performing home directory substitution home="/user/name" path="~/.aws/config" 
2024-03-27T19:33:56.501629Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/config" after="/user/name\\.aws\\config"
2024-03-27T19:33:56.501651Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file not found path=~/.aws/config
2024-03-27T19:33:56.501665Z DEBUG load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file loaded path=Some("/user/name\\.aws\\config") size=0
2024-03-27T19:33:56.501707Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: performing home directory substitution home="/user/name" path="~/.aws/credentials"
2024-03-27T19:33:56.501725Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/credentials" after="/user/name\\.aws\\credentials"
2024-03-27T19:33:56.501742Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file not found path=~/.aws/credentials
2024-03-27T19:33:56.501753Z DEBUG load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file loaded path=Some("/user/name\\.aws\\credentials") size=0     
2024-03-27T19:33:56.501788Z DEBUG load_config_file{file=FileContents { kind: Config, contents: "** redacted **" }}: aws_config::profile::parser::source: config file loaded path=None size=91
thread 'profile::parser::source::tests::programmatically_include_default_files' panicked at src\profile\parser\source.rs:447:9:
assertion `left == right` failed
  left: "[default]\nregion = us-east-1"
 right: ""

---- profile::credentials::test::credential_process_failure stdout ----
2024-03-27T19:33:56.478667Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}: aws_config::fs_util: loaded home directory src="HOME"
2024-03-27T19:33:56.478712Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/config"
2024-03-27T19:33:56.478796Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/config" after="/home\\.aws\\config"
2024-03-27T19:33:56.478921Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\config") size=101
2024-03-27T19:33:56.478970Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/credentials"
2024-03-27T19:33:56.478988Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/credentials" after="/home\\.aws\\credentials"
2024-03-27T19:33:56.479101Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process_failure\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\credentials") size=68
To see full logs from this test set VERBOSE_TEST_LOGS=true
test case: credential_process_failure. credential_process handles the external process exiting with a non-zero exit code
thread 'profile::credentials::test::credential_process_failure' panicked at src\test_case.rs:380:14:
`an error occurred while loading credentials: an error occurred while loading credentials: Error retrieving credentials: external process exited with code exit code: 1. Stderr:  (ProviderError(ProviderError { source: ProviderError(ProviderError { source: "Error retrieving credentials: external process exited with code exit code: 1. Stderr: " }) }))` did not contain `an error occurred while loading credentials: Error retrieving credentials: external process exited with code exit status: 1`

---- sso::cache::tests::determine_correct_cache_filenames stdout ----
thread 'sso::cache::tests::determine_correct_cache_filenames' panicked at src\sso\cache.rs:519:9:
assertion `left == right` failed
  left: "/home/someuser/.aws/sso/cache/d033e22ae348aeb5660fc2140aec35850c4da997.json"
 right: "/home/someuser\\.aws/sso/cache\\d033e22ae348aeb5660fc2140aec35850c4da997.json"

---- profile::credentials::test::credential_process stdout ----
2024-03-27T19:33:56.478380Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}: aws_config::fs_util: loaded home directory src="HOME"
2024-03-27T19:33:56.478421Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/config"
2024-03-27T19:33:56.478442Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/config" after="/home\\.aws\\config"
2024-03-27T19:33:56.478565Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Config)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\config") size=443
2024-03-27T19:33:56.478607Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: performing home directory substitution home="/home" path="~/.aws/credentials"
2024-03-27T19:33:56.478621Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: home directory expanded before="~/.aws/credentials" after="/home\\.aws\\credentials"
2024-03-27T19:33:56.478753Z DEBUG load_region{provider=ProfileFileRegionProvider { provider_config: ProviderConfig { env: Env(Fake({"HOME": "/home"})), fs: Fs(Fake(NamespacedFs { real_path: "./test-data/profile-provider/credential_process\\fs", namespaced_to: "/" })), time_source: SharedTimeSource(SystemTimeSource), http_client: Some(SharedHttpClient { selector: test_util::dvr::ReplayingClient }), sleep_impl: Some(SharedAsyncSleep(TokioSleep)), region: None, use_fips: None, use_dual_stack: None, profile_name_override: None } }}:load_config_file{file=Default(Credentials)}: aws_config::profile::parser::source: config file loaded path=Some("/home\\.aws\\credentials") size=68
To see full logs from this test set VERBOSE_TEST_LOGS=true
test case: credential_process. credential_process loads credentials from external process
thread 'profile::credentials::test::credential_process' panicked at src\test_case.rs:380:14:
expected output (Credentials { access_key_id: "ASIARTESTID", secret_access_key: "TESTSECRETKEY", session_token: Some("TESTSESSIONTOKEN"), expiry: Some(1651516560) }) but an error was returned: an error occurred while loading credentials: an error occurred while loading credentials: Error retrieving credentials: external process exited with code exit code: 255. Stderr: 'base64' is not recognized as an internal or external command,
operable program or batch file.
 (ProviderError(ProviderError { source: ProviderError(ProviderError { source: "Error retrieving credentials: external process exited with code exit code: 255. Stderr: 'base64' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n" }) }))

---- sso::cache::tests::save_cached_token stdout ----
2024-03-27T19:33:56.512492Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::cache::tests::save_cached_token' panicked at src\sso\cache.rs:561:14:
correct file written: Kind(NotFound)

---- sso::token::tests::expired_cached_token stdout ----
2024-03-27T19:33:56.512811Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.512832Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::expired_cached_token' panicked at src\sso\token.rs:498:13:
assertion failed: `str.contains(expected)`
expected: the SSO token has expired
     str: an error occurred while loading an access token: failed to load the cached SSO token (ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } }))


---- sso::token::tests::expired_token_and_expired_client_registration stdout ----
2024-03-27T19:33:56.513056Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.513074Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::expired_token_and_expired_client_registration' panicked at src\sso\token.rs:498:13:
assertion failed: `str.contains(expected)`
expected: the SSO token has expired
     str: an error occurred while loading an access token: failed to load the cached SSO token (ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } }))


---- sso::token::tests::expired_token_refresh_fails stdout ----
2024-03-27T19:33:56.513274Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.513287Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::expired_token_refresh_fails' panicked at src\sso\token.rs:467:18:
called `Result::unwrap()` on an `Err` value: ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } })

---- sso::token::tests::expired_token_refresh_with_refresh_token stdout ----
2024-03-27T19:33:56.513521Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.513535Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::expired_token_refresh_with_refresh_token' panicked at src\sso\token.rs:467:18:
called `Result::unwrap()` on an `Err` value: ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } })

---- sso::token::tests::expired_token_refresh_without_new_refresh_token stdout ----
2024-03-27T19:33:56.513802Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.513819Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::expired_token_refresh_without_new_refresh_token' panicked at src\sso\token.rs:467:18:
called `Result::unwrap()` on an `Err` value: ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } })

---- sso::token::tests::refresh_timings stdout ----
To see full logs from this test set VERBOSE_TEST_LOGS=true
thread 'sso::token::tests::refresh_timings' panicked at src\sso\token.rs:483:18:
called `Result::unwrap()` on an `Err` value: ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } })

---- sso::token::tests::use_unexpired_cached_token stdout ----
2024-03-27T19:33:56.514357Z DEBUG aws_config::sso::token: expiring cache asked for an updated SSO token
2024-03-27T19:33:56.514382Z DEBUG aws_config::fs_util: loaded home directory src="HOME"
thread 'sso::token::tests::use_unexpired_cached_token' panicked at src\sso\token.rs:483:18:
called `Result::unwrap()` on an `Err` value: ProviderError(ProviderError { source: FailedToLoadToken { source: IoError { what: "read", path: "/home/user\\.aws/sso/cache\\a94a8fe5ccb19ba61c4c0873d391e987982fbbd3.json", source: Kind(NotFound) } } })


failures:
    credential_process::test::test_credential_process
    credential_process::test::test_credential_process_no_expiry
    default_provider::credentials::test::eks_pod_identity_no_token_file
    default_provider::credentials::test::sso_no_token_file
    imds::client::test::one_second_connect_timeout
    imds::credentials::test::external_timeout_during_credentials_refresh_should_yield_last_retrieved_credentials
    profile::credentials::sso_tests::create_inner_provider_exactly_once
    profile::credentials::test::credential_process
    profile::credentials::test::credential_process_failure
    profile::parser::source::tests::programmatically_include_default_files
    sso::cache::tests::determine_correct_cache_filenames
    sso::cache::tests::save_cached_token
    sso::token::tests::expired_cached_token
    sso::token::tests::expired_token_and_expired_client_registration
    sso::token::tests::expired_token_refresh_fails
    sso::token::tests::expired_token_refresh_with_refresh_token
    sso::token::tests::expired_token_refresh_without_new_refresh_token
    sso::token::tests::refresh_timings
    sso::token::tests::use_unexpired_cached_token

test result: FAILED. 185 passed; 19 failed; 3 ignored; 0 measured; 0 filtered out; finished in 2.35s
```

**Review by ysaito1001 — APPROVED** [review]
[A list of test failures](https://github.com/smithy-lang/smithy-rs/pull/3522#issuecomment-2023816887) makes it easier for me to review and ensure we have added `#[cfg_attr(windows, ignore)]` to all target tests. Thanks for the fix!

File: codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt (hunk 1 above)
Comment (Velfi) [reply codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt]: TIL about `File.separator`. TY Aaron.

**Review by rcoh — APPROVED** [review]

**Pull request merged**